### ISSUE TITLE
Allow using Swup without transitions

### ISSUE THREAD

**Issue by danieldiekmeier** [swup/swup#510]
**Describe the problem**

I'm looking to transition some of my sites from Turbolinks to Swup. It only took a few minutes of work and it seems to work great – so no problem there.

The one thing that I was a bit suprised by is that Swup absolutely requires some elements with transitions. It says so right in the docs: https://swup.js.org/options#animation-selector

> As swup is built on animations, it is required to define the elements that are being animated.

But as I noticed, Swup does actually work perfectly fine without transitions – with one caveat. It prints a warning to the browser console:

> [swup] No animated elements found by selector [class*="transition-"]

This warning isn't helpful to me, because I personally don't want any transitions. (This might sound weird, but I'm really only looking to replace Turbolinks with another library. I'm not choosing Swup for its animation capabilities, but for its pjax-style browsing.)

**Describe the proposed solution**

My proposition is to allow setting `animationSelector: false` (or similar) to disable the search for transitioning elements. I think it could behave similarly like popstate events are handled:

https://github.com/swup/swup/blob/5749d7577f9fcbcd5c5292045304ceecccbadcd0/src/modules/loadPage.js#L50-L54

If this is an acceptable change, I'd be happy to submit a PR. Thanks for your consideration!

**Alternatives considered**

- Ignore the warnings – Don't really want to
- Add a dummy element with a transition that I don't really care about – If I'm not mistaken, this adds a small delay to page changes, which I don't want
- Patch the package with [`patch-package`](https://www.npmjs.com/package/patch-package) – I'll probably go with this for now

**How important is this feature to you?**

- [x] Nice to have
- [x] Would make my life a lot easier
- [ ] I cannot use swup without it

**Comment by daun** [comment]
Interesting! You're right that swup will work even without transitions. It wasn't meant to initially — until a few versions ago, swup would fail if no transitions were defined; we've only recenlty added the "fallback" to ease the support burden and gracefully handle incorrect setups.

Since swup advertises itself as a page transition library and the whole codebase is built on the assumption of handling transitions, I'd personally vote against making it optional. Let's see if the other two maintainers have an opinion on this.

In the meantime, there's two practical solutions I can think of:

1. Add a dummy transition, as you've mentioned. That's my recommendation. If you set it to 10ms, it shouldn't introduce a noticeable delay.
2. Remove console.log calls from your production bundles. That's assuming you don't mind the log output during development.

**Comment by mskocik** [comment]
And what about config option to turn "off" that transition warning?

**Comment by daun** [comment]
@mskocik That's actually a good idea, we might consider adding log verbosity levels to fine-tune the behavior. However, it's not something that's on the near horizon.

**Comment by gmrchk** [comment]
Actually, I think this is just a mistake we've made and I didn't realize recently. The scenario with no animated elements is generally supported and a valid use case I've used before myself. There is just no animation to wait on, so Swup proceeds without it.

The idea with the console.log was to let people know they are likely setting up the project incorrectly, because most people running into this situation did have a mistake in a setup and were surprised things don't animate. Those issues take a lot of time to go through and were very common, so we should keep the warning in place somehow, but leave the option to not get it when it's expected setup. 

The solutions for the time being proposed by @daun should work for sure, but to go ahead with this issue within Swup, I think we have two options: 
1. Adding option as originally proposed by the OP, so Swup knows this is a valid case. 
2. Using Swups `log` method, so the error would only be displayed with debug plugin. The plugin could also be improved to highlight such errors somehow, and adding a checkbox for "tested with debug plugin" into an issues template might also help for this.

**Comment by daun** [comment]
@gmrchk The more I think about it, allowing `false` as a selector is probably the most elegant solution. It signals intent, won't happen by accident and will be a simple check. No?

**Comment by gmrchk** [comment]
I agree, but mainly because it's the common issue we keep getting. This way we'd keep that warning for most users that actually make a mistake, while intentional setup would be supported. It's also a non-breaking API change, so why not. 
In other scenarios, I think we could use the debug plugin a bit more and provide all kinds of runtime checks.

**Comment by daun** [comment]
I've just merged #521 to allow passing `false` as the animation selector and suppress the warning. It'll be released as part of swup 3.0 which is currently taking shape in the `next` branch. Thanks for the suggestion!

**Comment by danieldiekmeier** [comment]
Thank you! That‘s great to hear. 😌
